src/luascript.cpp

```diff
@@ -5940,23 +5940,65 @@ int LuaScriptInterface::luaTileAddItem(lua_State* L)
 		}
 	}
 
-	uint32_t subType = tfs::lua::getNumber<uint32_t>(L, 3, 1);
+	const ItemType& it = Item::items[itemId];
 
-	Item* item = Item::CreateItem(itemId, std::min<uint32_t>(subType, ITEM_STACK_SIZE));
-	if (!item) {
+	int32_t itemCount = 1;
+	int32_t subType = 1;
+	uint32_t count = tfs::lua::getNumber<uint32_t>(L, 3, 1);
+
+	if (it.hasSubType()) {
+		if (it.stackable) {
+			itemCount = std::ceil(count / static_cast<float>(ITEM_STACK_SIZE));
```

Comment:
You can avoid the cast with

```suggestion
			itemCount = (count + ITEM_STACK_SIZE - 1) / ITEM_STACK_SIZE);
```

But it's not as clear and would need a comment